src/Parser/RetributionPaladin/Modules/PaladinCore/BoWProcTracker.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage, formatNumber } from 'common/format';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import Module from 'Parser/Core/Module';
+
+class BoWProcTracker extends Module {
+
+	overwrittenBoWProcs = 0;
+	totalBoWProcs = 0;
+
+	static dependencies = {
+		combatants: Combatants,
+	}
+
+	on_initialized(){
+		this.active = this.combatants.selected.hasTalent(SPELLS.BLADE_OF_WRATH_TALENT.id);
+	}
+	
+	on_byPlayer_applybuff(event){
+		const spellId = event.ability.guid;
+		if(spellId !== SPELLS.BLADE_OF_WRATH_PROC.id){
+			return;
+		}
+		this.totalBoWProcs++;
+	}
+
+	on_byPlayer_refreshbuff(event){
+		const spellId = event.ability.guid;
+		if(spellId !== SPELLS.BLADE_OF_WRATH_PROC.id){
+			return;
+		}
+		this.overwrittenBoWProcs++;
+		this.totalBoWProcs++;
+	}
+
+	suggestions(when) {
+		const missedProcsPercent = this.overwrittenBoWProcs / this.totalBoWProcs;
```

Comment:
Just a question, is there ever a time where this buff could expire without being used? If so you may want to include the `removebuff` trigger to account for that.

Reply:
I am currently not playing WoW anymore so I will have someone check for me but I am pretty sure it just instantly brings the spell off cooldown. The only time the buff can expire is if you get another proc without using the first one.